### PR TITLE
Feature/vtn 7176 file picker edit metadata

### PR DESCRIPTION
**Pull request by ChrisPelletier**
Added support to allow a user to pick a file for both program image and program live image image using the filepicker component.

### CODE REVIEW

File: packages/veritone-react-common/src/components/MediaInfoPanel/index.js [hunk 1]
```diff
@@ -360,20 +321,19 @@ class MediaInfoPanel extends Component {
                 Program Live Image
                 {get(
                   this.props.tdo,
-                  'details.veritoneProgram.signedProgramLiveImage.length',
+                  'details.veritoneProgram.programLiveImage.length',
```
Comment (AlexOVeritone) [inline packages/veritone-react-common/src/components/MediaInfoPanel/index.js]: That's ok as long as this uri is signed. The thing is that currently stored uri here are not signed.

File: packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ import TextField from 'material-ui/TextField';
 import { objectOf, func, bool, any } from 'prop-types';
 import { get } from 'lodash';
 import { withStyles } from 'material-ui/styles';
+import FilePicker from '../../FilePicker';
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js]: `import FilePicker from 'components/FilePicker'`

File: packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js [hunk 3]
```diff
@@ -68,13 +70,15 @@ class EditMetadataDialog extends Component {
...
+      metadataToUpdate.veritoneProgram.programLiveImage = this.state.programLiveImageFile;
     }
-    if (this.isProgramImageChanged()) {
-      metadataToUpdate.veritoneProgram = {};
-      metadataToUpdate.veritoneProgram.programImage = this.state.programImage;
+    if (this.state.programImageFile) {
+      if (!metadataToUpdate.veritoneProgram) {
+        metadataToUpdate.veritoneProgram = {};
+      }
+      metadataToUpdate.veritoneProgram.programImage = this.state.programImageFile;
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js]: I think you can reduce this to a one liner /w https://lodash.com/docs/4.17.5#setWith

`setWith(metadataToUpdate, 'veritoneProgram', this.state.programImageFile, Object);`

File: packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js [hunk 4]
```diff
@@ -118,6 +122,41 @@ class EditMetadataDialog extends Component {
...
+          selectFileType: null
+        };
+      });
+    };
+
+    fileReader.readAsDataURL(files[0]);
+  };
+
+  handleCloseFilePicker = () => {
+    this.setState((prevState, props) => {
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js]: don't need to use the cb here if you're not using prevState

File: packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js [hunk 5]
```diff
@@ -118,6 +122,41 @@ class EditMetadataDialog extends Component {
     );
   };
 
+  handleStartPickFiles = fileType => event => {
+    this.setState((prevState, props) => {
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js]: don't need to use the db here if you're not using prevState

File: packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import React, { Component } from 'react';
+import React, { Component, Fragment } from 'react';
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js]: all of the validation functions seem like unnecessary bloat, just do the comparison where you need the validation - no need to create a function for each one.

File: packages/veritone-widgets/src/redux/modules/mediaDetails/index.js [hunk 7]
```diff
@@ -242,13 +243,21 @@ export default createReducer(defaultState, {
...
       }
     };
   },
+  [UPDATE_TDO_ERROR](state, { payload, meta: { widgetId } }) {
+    const errorMessage = get(payload, 'message', payload);
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        updateTdoError: errorMessage ? errorMessage : null
```
Comment (charliemantle) [inline packages/veritone-widgets/src/redux/modules/mediaDetails/index.js]: errorMessage || null

File: packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js (hunk 3 above)
Comment (olslash) [reply packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js]:
```javascript
mediaToUpdate = {
  ...mediaToUpdate,
  veritoneProgram: {
    programImage: this.state.programImageFile
  }
}
```

File: packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js (hunk 6 above)
Comment (ChrisPelletier) [reply packages/veritone-react-common/src/components/MediaInfoPanel/EditMetadataDialog/index.js]: Resolved this in PR 138

**Pull request closed without merging**